src/swap/criptointercambio.js

```diff
@@ -0,0 +1,488 @@
+  return addressInfo.legacyAddress && !dontUseLegacy[currencyCode]
+    ? addressInfo.legacyAddress
+    : addressInfo.publicAddress
+}
+
+function checkReply(reply: Object, request: EdgeSwapRequest) {
+  if (reply.error != null) {
+    if (
+      reply.error.code === -32602 ||
+      /Invalid currency:/.test(reply.error.message)
+    ) {
+      throw new SwapCurrencyError(
+        swapInfo,
+        request.fromCurrencyCode,
+        request.toCurrencyCode
+      )
+    }
+    throw new Error('Criptointercambio error: ' + JSON.stringify(reply.error))
+  }
+}
+
+export function makeCriptointercambioPlugin(
+  opts: EdgeCorePluginOptions
+): EdgeSwapPlugin {
+  const { initOptions, io, log } = opts
+  const { fetchCors = io.fetch } = io
+
+  if (initOptions.apiKey == null || initOptions.secret == null) {
+    throw new Error('No Criptointercambio apiKey or secret provided.')
+  }
+  const { apiKey } = initOptions
+  const secret = parseUtf8(initOptions.secret)
+
+  async function call(json: any, promoCode?: string) {
+    const body = JSON.stringify(json)
+    const sign = base16
+      .stringify(hmacSha512(parseUtf8(body), secret))
+      .toLowerCase()
+
+    const headers: { [header: string]: string } = {
+      'Content-Type': 'application/json',
+      'api-key': apiKey,
+      sign
+    }
+    if (promoCode != null) headers['X-Promo-Code'] = promoCode
+    const response = await fetchCors(uri, { method: 'POST', body, headers })
+
+    if (!response.ok) {
+      throw new Error(`Criptointercambio returned error code ${response.status}`)
+    }
+    return response.json()
+  }
+
+  const out: EdgeSwapPlugin = {
+    swapInfo,
+    async fetchSwapQuote(
+      request: EdgeSwapRequest,
+      userSettings: Object | void,
+      opts: { promoCode?: string }
+    ): Promise<EdgeSwapQuote> {
+      checkInvalidCodes(INVALID_CURRENCY_CODES, request, swapInfo)
+
+      const fixedPromise = this.getFixedQuote(request, userSettings, opts)
+      // FIXME: Estimated swaps are temporarily disabled
+      const fixedResult = await fixedPromise
+      return fixedResult
+    },
+
+    async getFixedQuote(
+      request: EdgeSwapRequest,
+      userSettings: Object | void,
+      opts: { promoCode?: string }
+    ): Promise<EdgeSwapQuote> {
+      const { promoCode } = opts
+      const [fromAddress, toAddress] = await Promise.all([
+        getAddress(request.fromWallet, request.fromCurrencyCode),
+        getAddress(request.toWallet, request.toCurrencyCode)
+      ])
+      const quoteAmount =
+        request.quoteFor === 'from'
+          ? await request.fromWallet.nativeToDenomination(
+            request.nativeAmount,
+            request.fromCurrencyCode
+          )
+          : await request.toWallet.nativeToDenomination(
+            request.nativeAmount,
+            request.toCurrencyCode
+          )
+
+      const { safeFromCurrencyCode, safeToCurrencyCode } = safeCurrencyCodes(
```

Comment:
This is a bit concerning. Your native API should support specifying the chaincode separately from the token code. Without that, there could easily be confusion on what token the user wants to send or purchase. 

The CURRENCY_CODE_TRANSCRIPTION is a quick hack for prior partners but all new partners should have an explicit chaincode and tokencode. See the Changenow plugin

https://github.com/EdgeApp/edge-exchange-plugins/blob/master/src/swap/changenow.js#L110